core/src/main/java/io/questdb/preferences/PreferencesParser.java

```diff
@@ -0,0 +1,60 @@
+package io.questdb.preferences;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoException;
+import io.questdb.cutlass.json.AbstractJsonParser;
+import io.questdb.cutlass.json.JsonLexer;
+import io.questdb.std.CharSequenceObjHashMap;
+
+public final class PreferencesParser extends AbstractJsonParser {
+
+    private static final int STATE_ROOT = 1;
+    private static final int STATE_START = 0;
+    private final CharSequenceObjHashMap<CharSequence> parserMap;
+    private CharSequence key;
+    private int state = STATE_START;
+
+    public PreferencesParser(CairoConfiguration configuration, CharSequenceObjHashMap<CharSequence> parserMap) {
+        super(16384, configuration.getPreferencesParserCacheSizeLimit(), configuration.getPreferencesStringPoolCapacity());
```

Comment:
if 2 of 3 settings are taken from config, why the initial cache size is hardcoded?

Reply:
the thinking here was that instead of adding yet another config property (which will be barely used, if ever) we can just start with something sensible, and it will expand if it has to.